Address issue #166 - Change description and typical use case of countdown_pause_time (#167)
icpc/ccs-specs: Contest_API.md

| name                         | string          | Short display name of the contest.
| formal\_name                 | string ?        | Full name of the contest. Defaults to value of `name`.
| start\_time                  | TIME ?          | The scheduled start time of the contest, may be `null` if the start time is unknown or the countdown is paused.

> | countdown\_pause\_time       | RELTIME ?       | The amount of time left when the countdown to the contest start was paused, if the contest is paused, otherwise `null`.

Reviewer: ```suggestion
| countdown\_pause\_time       | RELTIME ?       | The amount of time left when the countdown to the contest start was paused, if the contest countdown is paused, otherwise `null`.
```

> | countdown\_pause\_time       | RELTIME ?       | The number of seconds left when the countdown to the contest start was paused. At no time may both `start_time` and `countdown_pause_time` be non-`null`.

Reviewer 2: But this is not a number, it's a `RELTIME`. The old wording; "amount of seconds" is clearly wrong, but maybe "amount of time" is a better fix than "number of seconds"?

There should be a "if the contest is paused" in there. Maybe something like this?:

```suggestion
| countdown\_pause\_time       | RELTIME ?       | The amount of time left when the countdown to the contest start was paused, if the contest is paused, otherwise `null`.
```

| logo                         | array of FILE ? | Logo for this contest, intended to be an image with aspect ratio near 1:1. Only allowed mime types are image/\*.
| location                     | LOCATION ?      | Location where the contest is held.

The typical use of `countdown_pause_time` is when a contest director wishes to pause the countdown to the start of a contest.  For example, this may occur because of technical
issues or to make an announcement.  When the contest countdown is paused, the value of `countdown_pause_time` should be set to the expected time remaining before the start of the contest after the pause is lifted.

> In this case, the `start_time` must be set to `null`, thereby, setting an undefined `start_time` (the `start_time` is undefined since

Reviewer: Clarify that we're referring to the `countdown_pause_time` attribute, not to the example use case in the previous sentences:
```suggestion
If `countdown_pause_time` is set, then the `start_time` must be set to `null`, thereby, setting an undefined `start_time` (the `start_time` is undefined since
```

the length of the pause may be unknown).

> The `countdown_pause_time` may be changed to indicate the approximate delay until the contest starts.

Reviewer 2: This is still *slightly* inconsistent (that's a strong word though) with the description. If the value is indeed "The number of seconds left when the countdown to the contest start was paused" then it couldn't be updated. 

Maybe we should say that this is "the amount of time left to the contest start when the contest is countdown is restarted. Typically set to the current amount of time remaining when countdown is paused"?

Author: I don't think the last part, "_Typically set to the current amount of time remaining when the countdown is paused_" adds anything except some confusion.  I had to read it a few times before I understood it, and I know what it is supposed to be.  In fact, after a conversation with @deboer-tim yesterday, the pause time is almost always NOT the time left when the contest was paused.  It's usually some "rounded" time close to the start of the contest.  The first part is better though...  So, perhaps something like this:
`When the contest countdown is paused, the value of countdown_pause_time should be set to the amount of time left to the expected contest start when then contest countdown is restarted.  (maybe "resumed" instead of "restarted"?)
`

Reviewer 2: > the pause time is almost always NOT the time left when the contest was paused

I don't think that's correct. The `countdown_pause_time` is almost always (as in: I can't think of a single instance when it has not been) the time left at the time it was paused. It is not uncommon though that this time is changed to something rounded, possibly very shortly after.

> perhaps something like this:
> When the contest countdown is paused, the value of countdown_pause_time should be set to the amount of time left to the expected contest start when then contest countdown is restarted. (maybe "resumed" instead of "restarted"?)

I wanted to avoid the implication that you have to know this at the time of pausing (because you would typically never know that).

Reviewer 3: As Fredrik said, the pause time is almost always set today by just 'stopping the clock', e.g. 3:05.57 remaining. Then after some announcements, it's often set to something like a 2:00 countdown, then a bit later the countdown will be resumed, by picking a time 2:00 in the future (rounded to the nearest second).

But all of this is based on a UI that was built for how we wanted to run at World Finals. There's no reason it couldn't pause at exactly 5:00, resume from 3:05.57, increase 'back' to 10:00, start at some arbitrary ms wall clock time, or "we're >4 minutes late already, lets just reschedule for 9:05am". It's purely a 'presentation value' to give the audience some idea of when the (currently unscheduled) contest might start. So, I think Fredrik is right that we shouldn't bake too many requirements or expectations into the spec based on how we currently use it.
